src/morphometrics/label/image_utils.py

```diff
@@ -0,0 +1,94 @@
+from typing import List, Optional, Union
+
+import numpy as np
+import pyclesperanto_prototype as cle
+
+from ..types import LabelImage
+
+
+def expand_selected_labels(
+    label_image: LabelImage,
+    label_values_to_expand: Union[int, List[int]],
+    expansion_amount: int = 1,
+    background_mask: Optional[np.ndarray] = None,
+) -> LabelImage:
+    """Expand selected label values in a label image.
+
+    Parameters
+    ----------
+    label_image : LabelImage
+        The label image to expand selected values in. Must be 2D or 3D.
+    label_values_to_expand : Union[int, List[int]]
+        The values in the label image that are allowed to expand.
+    expansion_amount : int
+        The radius of the expansion. Expansion is performed with equal
+        amounts in all directions.
+    background_mask : Optional[np.ndarray]
+        A boolean mask where True denotes background into which the labels
+        cannot be expanded. If None, the background mask is not applied.
+        Default value is None
+
+    Returns
+    -------
+    expanded_label_image : LabelImage
+        The resulting label image with the selected labels expanded.
+    """
+
+    if isinstance(label_values_to_expand, int):
+        # coerce into list
+        label_values_to_expand = [label_values_to_expand]
+
+    if label_image.ndim == 2:
+        label_image_is_2d = True
+        label_image = np.expand_dims(label_image, axis=0)
+    elif label_image.ndim == 3:
+        label_image_is_2d = False
+    else:
+        raise ValueError("label_image must be 2D or 3D")
+
+    gpu_labels = cle.push(label_image)
+
+    # preallocate the memory
+    flip = cle.create_labels_like(gpu_labels)
+    flop = cle.create_labels_like(gpu_labels)
+    flap = cle.create_labels_like(gpu_labels)
+    labels_mask = cle.create_labels_like(gpu_labels)
+    expanded_labels = cle.create_labels_like(gpu_labels)
+
+    kernel_size = cle.create([1, 1, 1])
```

Comment:
Just a little clarification: This variable is used further down as a `flag` to determine if the image was changed while processing it. It could be used to terminate the loop earlier. Just saying, the name "kernel_size" is misleading.

Reply:
Ah, thanks for the clarification. I think I misunderstood one of your examples 🤦